>[Performance Comparison (actions) [1/4] — Failed to parse the report.](https://s3.amazonaws.com/clickhouse-test-reports/33957/08f4f45fd9da923ae3e3fdd8a527c297d35247eb/performance_comparison__actions__[1/4]/runlog.log)

```
  0     0    0     0    0     0      0      0 --:--:--  0:00:19 --:--:--     0curl: (6) Could not resolve host: s3.amazonaws.com
```

>[Integration tests (release, actions) [2/2] — fail: 105, passed: 0, flaky: 0](https://s3.amazonaws.com/clickhouse-test-reports/33957/08f4f45fd9da923ae3e3fdd8a527c297d35247eb/integration_tests__release__actions__[2/2].html)

```
ERROR: for zoo2  Get "https://registry-1.docker.io/v2/": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```

```
E               ERROR: for node1  Get "https://registry-1.docker.io/v2/": context deadline exceeded
E               Get "https://registry-1.docker.io/v2/": context deadline exceeded
```

>[Stress test (address, actions) — Sanitizer assert (in stderr.log)](https://s3.amazonaws.com/clickhouse-test-reports/33957/08f4f45fd9da923ae3e3fdd8a527c297d35247eb/stress_test__address__actions_.html)

Interesting, more leaks had been found, but there were no leaks in the previous run, will take a look.

--- a/src/AggregateFunctions/AggregateFunctionSimpleState.h
+++ b/src/AggregateFunctions/AggregateFunctionSimpleState.h
@@ -35,18 +35,19 @@ public:
     {
         DataTypeCustomSimpleAggregateFunction::checkSupportedFunctions(nested_func);
 
-        // Need to make a clone because it'll be customized.
-        auto storage_type = DataTypeFactory::instance().get(nested_func->getReturnType()->getName());
-
+        // Need to make a clone to avoid recursive reference.
+        auto storage_type_out = DataTypeFactory::instance().get(nested_func->getReturnType()->getName());
         // Need to make a new function with promoted argument types because SimpleAggregates requires arg_type = return_type.
         AggregateFunctionProperties properties;
         auto function
-            = AggregateFunctionFactory::instance().get(nested_func->getName(), {storage_type}, nested_func->getParameters(), properties);
+            = AggregateFunctionFactory::instance().get(nested_func->getName(), {storage_type_out}, nested_func->getParameters(), properties);
 
+        // Need to make a clone because it'll be customized.
+        auto storage_type_arg = DataTypeFactory::instance().get(nested_func->getReturnType()->getName());
         DataTypeCustomNamePtr custom_name
             = std::make_unique<DataTypeCustomSimpleAggregateFunction>(function, DataTypes{nested_func->getReturnType()}, params);
-        storage_type->setCustomization(std::make_unique<DataTypeCustomDesc>(std::move(custom_name), nullptr));
-        return storage_type;
+        storage_type_arg->setCustomization(std::make_unique<DataTypeCustomDesc>(std::move(custom_name), nullptr));
+        return storage_type_arg;
     }
 
     bool isVersioned() const override

--- a/src/Interpreters/executeQuery.cpp
+++ b/src/Interpreters/executeQuery.cpp
@@ -61,6 +61,7 @@
 #include <Processors/Sources/WaitForAsyncInsertSource.h>
 
 #include <base/EnumReflection.h>
+#include <base/demangle.h>
 
 #include <random>
 
@@ -659,7 +660,7 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
                 if (context->query_trace_context.trace_id != UUID())
                 {
                     auto * raw_interpreter_ptr = interpreter.get();
-                    std::string class_name(abi::__cxa_demangle(typeid(*raw_interpreter_ptr).name(), nullptr, nullptr, nullptr));
+                    std::string class_name(demangle(typeid(*raw_interpreter_ptr).name()));
                     span = std::make_unique<OpenTelemetrySpanHolder>(class_name + "::execute()");
                 }
                 res = interpreter->execute();

--- a/tests/queries/0_stateless/01570_aggregator_combinator_simple_state.reference
+++ b/tests/queries/0_stateless/01570_aggregator_combinator_simple_state.reference
@@ -1,14 +1,31 @@
+-- { echo }
+with anySimpleState(number) as c select toTypeName(c), c from numbers(1);
 SimpleAggregateFunction(any, UInt64)	0
+with anyLastSimpleState(number) as c select toTypeName(c), c from numbers(1);
 SimpleAggregateFunction(anyLast, UInt64)	0
+with minSimpleState(number) as c select toTypeName(c), c from numbers(1);
 SimpleAggregateFunction(min, UInt64)	0
+with maxSimpleState(number) as c select toTypeName(c), c from numbers(1);
 SimpleAggregateFunction(max, UInt64)	0
+with sumSimpleState(number) as c select toTypeName(c), c from numbers(1);
 SimpleAggregateFunction(sum, UInt64)	0
+with sumWithOverflowSimpleState(number) as c select toTypeName(c), c from numbers(1);
 SimpleAggregateFunction(sumWithOverflow, UInt64)	0
+with groupBitAndSimpleState(number) as c select toTypeName(c), c from numbers(1);
 SimpleAggregateFunction(groupBitAnd, UInt64)	0
+with groupBitOrSimpleState(number) as c select toTypeName(c), c from numbers(1);
 SimpleAggregateFunction(groupBitOr, UInt64)	0
+with groupBitXorSimpleState(number) as c select toTypeName(c), c from numbers(1);
 SimpleAggregateFunction(groupBitXor, UInt64)	0
+with sumMapSimpleState(([number], [number])) as c select toTypeName(c), c from numbers(1);
 SimpleAggregateFunction(sumMap, Tuple(Array(UInt64), Array(UInt64)))	([],[])
+with minMapSimpleState(([number], [number])) as c select toTypeName(c), c from numbers(1);
 SimpleAggregateFunction(minMap, Tuple(Array(UInt64), Array(UInt64)))	([0],[0])
+with maxMapSimpleState(([number], [number])) as c select toTypeName(c), c from numbers(1);
 SimpleAggregateFunction(maxMap, Tuple(Array(UInt64), Array(UInt64)))	([0],[0])
+with groupArrayArraySimpleState([number]) as c select toTypeName(c), c from numbers(1);
 SimpleAggregateFunction(groupArrayArray, Array(UInt64))	[0]
+with groupUniqArrayArraySimpleState([number]) as c select toTypeName(c), c from numbers(1);
 SimpleAggregateFunction(groupUniqArrayArray, Array(UInt64))	[0]
+-- non-SimpleAggregateFunction
+with countSimpleState(number) as c select toTypeName(c), c from numbers(1); -- { serverError 36 }

--- a/tests/queries/0_stateless/01570_aggregator_combinator_simple_state.sql
+++ b/tests/queries/0_stateless/01570_aggregator_combinator_simple_state.sql
@@ -1,3 +1,4 @@
+-- { echo }
 with anySimpleState(number) as c select toTypeName(c), c from numbers(1);
 with anyLastSimpleState(number) as c select toTypeName(c), c from numbers(1);
 with minSimpleState(number) as c select toTypeName(c), c from numbers(1);

